src/classes/HealthCheckGroupService.cls

```diff
@@ -0,0 +1,106 @@
+    * @description Instance for Singleton Pattern
+    **********************************************/
+    @TestVisible
+    private static HealthCheckGroupService instance;
+
+    /*********************************************************************
+    * @description Empty constructor is not public for Singleton Pattern
+    **********************************************************************/
+    protected HealthCheckGroupService() {}
+
+    /******************************************************************************
+    * @description Static method to get the current instance for Singleton pattern
+    * @return The instance of HealthCheckGroupService.
+    ******************************************************************************/
+    public static HealthCheckGroupService getInstance() {
+
+        if (instance == null) {
+            instance = new HealthCheckGroupService();
+        }
+
+        return instance;
+    }
+
+    /********************************************************************************************
+    * @description Retrieves the HealthCheckGroup view model for a specified 
+    * HealthCheckDefinitionModel.
+    * @param  name The name corresponding to a specified HealthCheckDefinitionModel
+    * @param  className The className corresponding to a specified HealthCheckDefinitionModel
+    * @param  namespace The namespace corresponding to a specified HealthCheckDefinitionModel
+    * @return A HealthCheckGroupVModel representing the specified HealthCheckDefinitionModel.
+    ********************************************************************************************/
+    public virtual HealthCheckGroupVModel getHealthCheckGroup(String name, String className, String namespace){
+        HealthCheckGroupAPIServiceInterface hlthChkGroupServInterface = 
+            this.locateHealthCheckGroupAPIService(className, namespace);
+
+        return hlthChkGroupServInterface.getHealthCheckGroup(name);
+    }
+
+    /********************************************************************************************
+    * @description Dynamically instantiates a HealthCheckGroupAPIServiceInterface based on a 
+    * specified classname and namespace.
+    * @param  className The classname of a HealthCheckDefinitionModel
+    * @param  namespace The namespace of a HealthCheckDefinitionModel
+    * @return A HealthCheckGroupAPIServiceInterface corresponding to the specified classname and namespace.
+    ********************************************************************************************/
+    @TestVisible
+    private HealthCheckGroupAPIServiceInterface locateHealthCheckGroupAPIService(String className, String namespace){
+        HealthCheckGroupAPIServiceInterface hlthChkGroupServInterface;
+        String namespacedClassName = className;
+
+        if (String.isNotBlank(namespace) && !namespace.equalsIgnoreCase('hed')){
+            namespacedClassName = namespace + '.' + className;
```

Comment:
Please make a test for this line. You should be able to use the null pointer exception test to cover it.